### DISCUSSION TITLE
How can new readers or extensions be added?

### DISCUSSION THREAD

**Discussion by scarrick68** [ericblade/quagga2#493] [Q&A]
I would like to be able to read any 1-dimensional barcode string without having to specify the reader type. Is this possible?
I am running into what appears to be some non-standard barcode types where some manufacturers just have custom barcodes. They look like normal 1-D barcodes but with non-standard lengths like 8 numbers. UPC and EAN variants do not decode them correctly.

**Comment by ericblade** [comment]
If you run every reader in the same run, it will potentially lead to erroneous decodes, as some readers are subsets of others.  You could build a front-end that takes in the images, and then calls Quagga.decodeSingle() with each reader, and returns the results of all of them.  You could streamline that by taking some time to figure out which readers can have conflicting results with other readers (could be a research project on it's own), and running it in groups that don't include multiples that can read the same code.

I think almost all of users use cases tend to revolve around one or two different types of barcodes, so it's not something i want to try to support as a primary use case .. although if/when my plans for a 2.0 interface ever materialize, it shouldn't be difficult at all.

but it can be done right now, using decodesingle.
